### DISCUSSION TITLE
Easy Transcription, a transcriber for audios and videos on Android

### DISCUSSION THREAD

**Discussion by Digipom** [ggml-org/whisper.cpp#1266] [Show and tell]
We've released our first beta of Easy Transcription for transcribing audio and video directly on your Android device. Easy Transcription currently supports the following features:
* Open audio and video from inside the app using any of Android's supported formats (such as MP3, MP4 and PCM WAVE), and then transcribe them, or translate them to English. You can also share to the app from other apps.
* Edit, search & replace from within a transcription.
* Copy and share your transcriptions in different formats.

We support tiny and tiny.en for now, with support for more models coming soon. Everything is transcribed privately directly on the device. We're also planning to add a built-in voice recorder in a future release. Please don't hesitate to send us feedback here or inside the app via About > Talk to us > By email.

The app was implemented using Android Compose and uses whisper.cpp to handle the transcriptions. Many thanks go to @ggerganov and all of the other contributors as well for creating and working on whisper.cpp -- it's amazing that we now have a great open-source library for doing transcriptions right on the device!

You can download the app for free at https://play.google.com/store/apps/details?id=com.digipom.easytranscription
<p align="middle">
<img src="https://github.com/ggerganov/whisper.cpp/assets/2005192/6a3ca7d4-76b6-4977-80de-0b8ef9dabe1d" width=25% height=25%>
<img src="https://github.com/ggerganov/whisper.cpp/assets/2005192/e560f270-bde1-474c-9e60-3e274bca0b84" width=25% height=25%>
<img src="https://github.com/ggerganov/whisper.cpp/assets/2005192/5c5c6252-d6ee-4c13-90bc-53a416191f4c" width=25% height=25%>

**Comment by kwanUm** [comment]
Is this app open sourced?

**Reply by Digipom** [reply]
It's not, but it builds upon the Android sample which is: https://github.com/ggerganov/whisper.cpp/tree/master/examples/whisper.android

**Reply by ducthang1752k4-ctrl** [reply]
i cant install it from CHplay any more

**Comment by LumenYoung** [comment]
This is a really great application! I wonder how this would go because I see it is been constantly updated, yet no clear whether it is to open source or commercialize? Thanks

**Reply by Digipom** [reply]
Thank you! For now the intent is to commercialize it, though some aspects are open-sourced via this repository and also the Android code here.

**Comment by SatoX69** [comment]
Can you try porting or making an app with Suno's Bark model, I reckon there's a bark.cpp repository

**Comment by MonsterSe7en** [comment]
Hey, is there a way to record audio on my Android device, specify the language and Whisper model on the phone itself, and then have my PC handle the transcription while keeping the same settings from my Android?

**Reply by Digipom** [reply]
For this, probably easiest way is to use any voice recorder to record to an audio file and then just stream that to the PC using something like MetaCtrl Sync.

**Comment by LaptopDev** [comment]
How did you manage long context recording splits? It is very accurate, even if there may be hallucinations some times on long pauses. 

I can't get the same consistency with the args I am using on my pc and it is unreliable on basically any audio length:
```
    # long: VAD
    out_srt=$("$WHISPER_BIN" \
      -m "$WHISPER_MODEL" \
      -f "$file" \
      --vad \
      --vad-model "$VAD_MODEL" \
      --vad-threshold 0.6 \
      --vad-min-silence-duration-ms 200

```
I am using ggml-silero-v5.1.2.bin

**Reply by Digipom** [reply]
We split it into frames of 5 minutes each. It doesn't look like we preserve the previous context for the next slice currently.

Whisper is initialized with whisper_full_default_params(WHISPER_SAMPLING_GREEDY); and then we call whisper_full. On the phone we use either ggml_tiny.en.bin or ggml_base.bin.

**Comment by quanvm31** [comment]
do you face the problems when there is no actual talking, just the audio with the sound of knocking or sth, whisper still gene the hallucniation text?

**Comment by khizar-anjum** [comment]
Hi, is this app no longer available? I looked at the url at https://play.google.com/store/apps/details?id=com.digipom.easytranscription but it says "not found". Any help would be greatly appreciated!

**Reply by Digipom** [reply]
The app is currently unpublished due to the realities of on-device transcription across a wide variety of Android devices and languages and a lack of resources on my end to maintain multiple apps. However, I'm planning to bring the feature back in the future. Please send me a direct message if you'd like more details (best way is contact @ my domain.com).

**Reply by khizar-anjum** [reply]
I see, I was thinking of making a similar app for note taking on android that is fully local. I like the idea, might be worth it to contribute and help you instead since you already have an app going. Will contact you over the weekend.

**Comment by RafiMuhluvch33s3** [comment]
When will you re-publish this app again? I never tried it before.

**Reply by Digipom** [reply]
No plans right now, but we're planning to release the functionality as part of EVR this year. In the meantime whisper.cpp also gives good functionality if you copy the files to your PC first.

**Reply by RafiMuhluvch33s3** [reply]
C'mon man. I wanna try it. Just re-open it.

**Comment by prashantweb3task-collab** [comment]
This looks really promising! Having a [Transcription app](https://voicetonotes.ai/blog/voice-to-notes/) that processes audio entirely on the device is a big advantage for privacy, especially for users who don't want to upload recordings to the cloud. I also like that it supports editing and exporting transcripts directly within the app. Looking forward to seeing support for larger models and the built-in voice recorder in future updates. Great work on the first beta!

**Comment by shivangiy1610-coder** [comment]
Nice work! It's great to see another app using Whisper for on-device transcription. Privacy-first processing is something more users are starting to value.

I've been testing a few transcription apps recently, including [Voice Note App](https://voicetonotes.ai/blog/voice-to-notes/), and it's interesting how much the post-transcription editing experience affects usability. Looking forward to trying future updates, especially the larger models.
